src/SignalR/server/StackExchangeRedis/test/RedisHubLifetimeManagerTests.cs

```diff
@@ -83,6 +86,48 @@ public async Task CamelCasedJsonIsPreservedAcrossRedisBoundary()
         }
     }
 
+    [Fact]
+    public async Task ErrorFromConnectionFactoryLogsAndAllowsDisconnect()
+    {
+        var server = new TestRedisServer();
+
+        var testSink = new TestSink();
+        var logger = new TestLogger("", testSink, true);
+        var mockLoggerFactory = new Mock<ILoggerFactory>();
+        mockLoggerFactory
+            .Setup(m => m.CreateLogger(It.IsAny<string>()))
+            .Returns((string categoryName) => (ILogger)logger);
+        var loggerT = mockLoggerFactory.Object.CreateLogger<RedisHubLifetimeManager<Hub>>();
+
+        var manager = new RedisHubLifetimeManager<Hub>(
+            loggerT,
+            Options.Create(new RedisOptions()
+            {
+                ConnectionFactory = _ => throw new Exception("throw from connect")
+            }),
+            new DefaultHubProtocolResolver(new IHubProtocol[]
+            {
+            }, NullLogger<DefaultHubProtocolResolver>.Instance));
+
+        using (var client = new TestClient())
+        {
+            var connection = HubConnectionContextUtils.Create(client.Connection);
+
+            var ex = await Assert.ThrowsAsync<Exception>(() => manager.OnConnectedAsync(connection)).DefaultTimeout();
+            Assert.Equal("throw from connect", ex.Message);
+
+            await manager.OnDisconnectedAsync(connection).DefaultTimeout();
+        }
+
+        var logs = testSink.Writes.ToArray();
+        // Two of the same error logs, one for the pre-emptive attempt to connect to Redis in the ctor, and one during the OnConnectedAsync for the connection
+        Assert.Equal(2, logs.Length);
+        Assert.Equal("Error connecting to Redis.", logs[0].Message);
+        Assert.Equal("throw from connect", logs[0].Exception.Message);
+        Assert.Equal("Error connecting to Redis.", logs[1].Message);
+        Assert.Equal("throw from connect", logs[1].Exception.Message);
```

Comment:
Do we really want to repeat this log every time a client connects or any time the app tries to publish a message? Would it be better to use a flag to see if we've already logged `ErrorConnecting` and not log again? I think the first error should be sufficient in most cases. And with the exception of the first call made in the constructor which we'd log, we're rethrowing the exception anyway.

 I'm okay with